### PR TITLE
Update environment.py

### PR DESCRIPTION
**Pull request by aothms**

### CODE REVIEW

**Review by JTurbakiewicz — APPROVED** [review]

**Comment by JTurbakiewicz** [comment]
I'm not sure if context.config.userdata["input"] will ever return something that's not hashable, but I think should be fine.

**Review by Ghesselink — APPROVED** [review]
Nice :) Checked it locally on a single file with the debugger as well. As expected, the cache gets saved nicely in the 'model_cache = {}' during the first iteration and is then retrieved in subsequent iterations

Is it also something to use in the validation server? We're now opening, if I'm correct, the file at least 5 times (at info.py, gherkin, bsdd, mvd, validate). But I guess that is more tricky because we are using redis/queue. Maybe somewhere in the database before all the tasks are executed? 
We can discuss that tomorrow .. (I'll also have to get a more in depth understanding of how everything in the validation service is tied together. )

File: features/environment.py [hunk 1]
```diff
@@ -2,14 +2,21 @@
 
 from behave.model import Scenario
 
+model_cache = {}
+def read_model(fn):
+    if cached := model_cache.get(fn):
+        return cached
```
[the remark this answers is not in the document]
Comment (aothms) [reply features/environment.py:8]: yes, these are one of the few cases where the walrus makes sense

**Comment by aothms** [comment]
> Is it also something to use in the validation server? We're now opening, if I'm correct, the file at least 5 times (at info.py, gherkin, bsdd, mvd, validate)

> But I guess that is more tricky because we are using redis/queue

It's not so much related to the redis/queuing mechanism because all these 5 operations are performed within a single task of the task queue. The difference is that we have modular scripts that take care of all specific operations and you can't transfer a parsed file to another process.

It's also not really an issue there because these operations are not ran in parallel so the same file won't be open multiple times simultaneously (which was the case here). And parsing the file itself isn't that significant wrt to the overall process of these 5 operations (well it is for general info and a small feature test, but not for schema). In this case I find the modularity of the code of higher value.

**Pull request merged**